### ISSUE TITLE
Double delete of vmdk when PV gets deleted.

### ISSUE THREAD

**Issue by mattmattox** [rancher/rancher#26933]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
When deleting a PV, the VMDK is deleted but the delete task is called twice. The first task successes but the second task fails because the VMDK is already deleted.
- Create PV using the default vsphere storageclass
- Delete PV
- Watch logs in vCenter

**Result:**
Upstream bug: [90770](https://github.com/kubernetes/kubernetes/issues/90770)

**Other details that may be helpful:**
Kubernetes version (use kubectl version): v1.15.5
Cloud provider or hardware configuration: On-perm VMware
OS (e.g: cat /etc/os-release): Ubuntu 16.04.04

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.3.6
- Installation option (single install/HA): HA

**Comment by leodotcloud** [comment]
I am not sure if the successive delete operations are triggered by k8s. The delete operations have the same opID with differing "sub" field. The same opID indicates that the second delete operation did not initiate from rancher/k8s but it could be some internal task triggered inside vCenter.


DELETE SUCCESS:
var/run/log/hostd.log:2020-06-19T00:08:31.300Z info hostd[2796274] [Originator@6876 sub=OsfsClient opID=15f91743-46-f3bc user=vpxuser:VSPHERE.LOCAL\someaccountname] Done deleting objects in directory: /vmfs/volumes/abcdefg-312344b4-1234-xyz628c87f90/kubevols/kubernetes-dynamic-pvc-12345abc-abcd-abcd-bfbb-abcd124axyzy.vmdk

DELETE FAILURE:
var/run/log/hostd.log:2020-06-19T00:08:31.303Z error hostd[2796274] [Originator@6876 sub=Nfcsvc opID=15f91743-46-f3bc user=vpxuser:VSPHERE.LOCAL\someaccountname] file delete force (/vmfs/volumes/abcdefg-312344b4-1234-xyz628c87f90/kubevols/kubernetes-dynamic-pvc-12345abc-abcd-abcd-bfbb-abcd124axyzy.vmdk -> /vmfs/volumes/5e14acac-3cfcc4b4-ea47-b02628c87f90/kubevols/kubernetes-dynamic-pvc-12345abc-abcd-abcd-bfbb-abcd124axyzy.vmdk) by 'vpxuser'

**Comment by leodotcloud** [comment]
Shared my findings on the upstream issue. See [here](https://github.com/kubernetes/kubernetes/issues/90770#issuecomment-653940492).

TL;DR:
The second delete operation results in an error message in the kube-controller-manager logs, but it's harmless. Doesn't cause an inconsistent state.
